text/3263-pre-release-sticky.md

```diff
@@ -0,0 +1,115 @@
+- Feature Name: `pre-release-sticky`
+- Start Date: 2022-05-10
+- RFC PR: [rust-lang/rfcs#3263](https://github.com/rust-lang/rfcs/pull/3263)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+The following assume:
+
+ - A release is a version without a "pre-release" tag (like `1.0.0`)
+ - A pre-release is a version with a "pre-release" tag (like `1.0.0-alpha`)
+
+Cargo will now apply no precedence rule on pre-release by default, this mean that `1.0.0-alpha.0` will be interpreted as `=1.0.0-alpha.0` by cargo instead of `^1.0.0-alpha.0`.
+
+# Motivation
+[motivation]: #motivation
+
+The current behavior often break pre-release build, [Semver 2.0](https://semver.org/#spec-item-11) rules for pre-release doesn't include breaking change concept between pre-release. This mean that any new pre-release would be considerate by cargo as a "compatible update". Even worse, final version also match the requierement of any pre-release. Say otherwise, if an user put `version = "1.0.0-alpha.0"` in their `Cargo.toml` this will be considered by cargo as compatible with any pre-release that follow `1.0.0-alpha.1`, `1.0.0-alpha.2`, `1.0.0-beta.0`, etc... and also `1.0.0`, `1.1.0`, etc... (But not `2.0.0`). This can also lead to security problem.
```

Comment:
> Precedence rules seem to be about ordering version tags, whereas the issue here is not about ordering but about when Cargo considers a version to be compatible with another version.

That precisely the purpose of precedence rules.

> I'd try to make this section a bit less opinionated ("often break") and be more factual. Would be interesting to add more supporting references for why this problem is big enough to bother, so I would lead with the Cargo issue and internals thread here.

Will try